numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,35 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape=None):
+    """
+    Create rolling window views of the 2D array with the given shape.
+
+    .. warning:: This function has to be used with extreme care, see notes from 'as_strided'.
+
+    For some cases, there may be more efficient approaches
+
+    Parameters
+    ----------
+    x : ndarray
+        2D array to create rolling window views.
+    shape : sequence of int, optional
+        The shape of the new array. Defaults to ``x.shape``.
```

Comment:
This is not accurate - it's the shape of the window, right?

This default doesn't seem useful

Reply:
I use the same default setting as it is in 'as_strided'. Any change suggestion?

Reply:
Remove the default entirely

Reply:
Shape default removed.